runtime/src/UrlUtils.ts

```diff
@@ -0,0 +1,22 @@
+export function parseExperienceURL(
+  experienceURL: string
+): { channel: string; testTransport: string | null } | null {
+  const matches = experienceURL.match(/(\+|\/sdk\..*-)([^?]*)\??(.*$)/);
+  if (!matches) {
+    return null;
+  }
+  const channel = matches[2];
+
+  let testTransport = null;
+  const queryItems = (matches[3] ?? '').split(/&/g);
+  for (const item of queryItems) {
+    if (item.startsWith('testTransport=')) {
+      testTransport = item.substring(14);
+      break;
+    }
+  }
```

Comment:
Using the built-in URL API for parsing might be easier. It has been supported for about 7-8 years: https://caniuse.com/?search=URL